python/GafferImageUI/ImageInspector.py

```diff
@@ -0,0 +1,460 @@
+		self.__formatPathListing.setPath(
+			_FormatPath( self.scriptNode(), self.__settingsNode["__selectView"]["out"], context, "/" ),
+		)
+		self.__channelsPathListing.setPath(
+			_ChannelPath( self.scriptNode(), self.__settingsNode["__imageStats"]["in"], context, "/" ),
+		)
+		self.__metadataPathListing.setPath(
+			_MetadataPath( self.scriptNode(), self.__settingsNode["__selectView"]["out"], context, "/" ),
+		)
+
+GafferUI.Editor.registerType( "ImageInspector", ImageInspector )
+
+##########################################################################
+# Metadata controlling the settings UI
+##########################################################################
+
+Gaffer.Metadata.registerNode(
+
+	ImageInspector.Settings,
+
+	plugs = {
+
+		"view" : [
+
+			"plugValueWidget:type", "GafferImageUI.ViewPlugValueWidget",
+			"description", "The view to inspect",
+
+		],
+
+	}
+
+)
+
+##########################################################################
+# _ImagePath
+##########################################################################
+
+class _ImagePath( Gaffer.Path ) :
+
+	def __init__( self, scriptNode, image, context, path, root = "/", filter = None ) :
+
+		Gaffer.Path.__init__( self, path, root, filter )
+
+		self.__scriptNode = scriptNode
+		self._context = context
+		self._image = image
+		self.__imagePlugDirtiedConnection = self._image.node().plugDirtiedSignal().connect(
+			Gaffer.WeakMethod( self.__imagePlugDirtied, fallbackResult = None ), scoped = True
+		)
+
+	def copy( self ) :
+
+		return self.__class__( self.__scriptNode, self._image, self._context, self[:], self.root(), self.getFilter() )
+
+	def cancellationSubject( self ) :
+
+		## \todo We really just want to return `self._image` here, but we
+		# can't because BackgroundTask can't find the right ScriptNode from it.
+		return self.__scriptNode["fileName"]
+
+	def _children( self, canceller ) :
+
+		return [
+			self.__class__( self.__scriptNode, self._image, self._context, self[:] + [ name ], self.root(), self.getFilter() )
+			for name in self._childNames( canceller )
+		]
+
+	def _childNames( self, canceller ) :
+
+		raise NotImplementedError
+
+	def __imagePlugDirtied( self, plug ) :
+
+		if plug.parent() == self._image and self._imagePlugAffectsPath( plug ) :
+			self._emitPathChanged()
+
+##########################################################################
+# _FormatPath
+##########################################################################
+
+class _FormatPath( _ImagePath ) :
+
+	def __init__( self, scriptNode, image, context, path, root = "/", filter = None ) :
+
+		_ImagePath.__init__( self, scriptNode, image, context, path, root, filter )
+
+	def propertyNames( self, canceller = None ) :
+
+		return Gaffer.Path.propertyNames( self ) + [ "format:value" ]
+
+	def property( self, name, canceller = None ) :
+
+		if name != "format:value" :
+			return Gaffer.Path.property( self, name, canceller )
+
+		if len( self ) != 1 :
```

Comment:
Could you clarify what case this is catching?

Reply:
Paths can be any length, with `len( self ) == 0` denoting the root path, and paths where `len( self ) > 1` being invalid (since they could never be created from `_childNames()`). We only have properties on the valid paths.